omod/src/main/webapp/resources/scripts/conditionlist/controllers/addcondition.controller.js

```diff
@@ -31,34 +36,62 @@ function ConditionController($scope, RestfulService, ConditionModel, ConceptMode
                 emr.errorAlert("Error Saving condition");
             });
         }
+    }
 
     self.initCondition = self.initCondition || function () {
             $scope.patientUuid = CommonFunctions.extractUrlArgs(window.location.search)['patientId'];
             $scope.condition = new ConditionModel($scope.patientUuid);
+            self.showEndDate();
         }
 
     self.validateCondition = self.validateCondition || function () {
-            var concept = $scope.concept.concept;
-            if (concept !== null) {
-                $scope.condition.concept = new ConceptModel(concept.uuid, concept.display);
-            } else {
-                // noncoded diagnosis
-                $scope.condition.conditionNonCoded = "NON_CODED:" + $scope.concept.word;
-            }
+        var concept ;
+        if($scope.concept.concept.uuid) {
+            concept=$scope.concept.concept;
+        } else {
+            concept=$scope.condition.concept;
+        }
+
+        if (concept !== null) {
+            $scope.condition.concept = new ConceptModel(concept.uuid, concept.display);
+        } else {
+            // noncoded diagnosis
+            $scope.condition.conditionNonCoded = "NON_CODED:" + $scope.concept.word;
+        }
 
-            $scope.condition.onSetDate = self.getSelectedDate();
+        $scope.condition.onSetDate = self.getSelectedDate();
 
-            self.saveCondition();
+        if($scope.condition.status==INACTIVE_STATUS) {
+            $scope.condition.endDate = self.getEndDate();
         }
+        self.saveCondition();
+    }
+
+    self.showEndDate = self.showEndDate || function() {
+        var groups = document.getElementsByClassName("group");
+        if($scope.condition.status==INACTIVE_STATUS){
+            groups[2].style.visibility = "visible";
+        }
+        else{
+            groups[2].style.visibility = "hidden";    
+        }
+     }        
 
     self.getSelectedDate = self.getSelectedDate || function () {
             var datePicker = angular.element(document.getElementsByName('conditionStartDate'))[0];
             return datePicker.value;
         }
 
+    self.getEndDate = self.getEndDate || function() {
+            var endDatePicker = angular.element(document.getElementsByName('conditionEndDate'))[0];
```

Comment:
Do we need a temporary variable just for the sake of calling .value on it?

Reply:
I think it's more clear to write it this way. Also getSelectedDate()  has been written in the same fashion